app/src/org/commcare/models/database/IndexedFixturePathUtils.java

```diff
@@ -19,34 +24,65 @@
-        Cursor c = db.query(IndexedFixturePathsConstants.INDEXED_FIXTURE_PATHS_TABLE,
-                new String[]{IndexedFixturePathsConstants.INDEXED_FIXTURE_PATHS_COL_BASE, IndexedFixturePathsConstants.INDEXED_FIXTURE_PATHS_COL_CHILD},
-                IndexedFixturePathsConstants.INDEXED_FIXTURE_PATHS_COL_NAME + "=?", new String[]{fixtureName}, null, null, null);
+    public final static String INDEXED_FIXTURE_PATHS_TABLE = "IndexedFixtureIndex";
+    public final static String INDEXED_FIXTURE_PATHS_COL_NAME = "name";
+    public final static String INDEXED_FIXTURE_PATHS_COL_BASE = "base";
+    public final static String INDEXED_FIXTURE_PATHS_COL_CHILD = "child";
+    public final static String INDEXED_FIXTURE_PATHS_COL_ATTRIBUTES = "attributes";
+
+    public final static String INDEXED_FIXTURE_PATHS_TABLE_STMT =
+            "CREATE TABLE IF NOT EXISTS " +
+                    INDEXED_FIXTURE_PATHS_TABLE +
+                    " (" + INDEXED_FIXTURE_PATHS_COL_NAME + " UNIQUE" +
+                    ", " + INDEXED_FIXTURE_PATHS_COL_BASE +
+                    ", " + INDEXED_FIXTURE_PATHS_COL_CHILD +
+                    ", " + INDEXED_FIXTURE_PATHS_COL_ATTRIBUTES + ");";
+
+    public final static String INDEXED_FIXTURE_INDEXING_STMT =
+            DatabaseIndexingUtils.indexOnTableCommand("fixture_name_index",
+                    INDEXED_FIXTURE_PATHS_TABLE, INDEXED_FIXTURE_PATHS_COL_NAME);
+
+    public final static String INDEXED_FIXTURE_PATHS_TABLE_STMT_V15 =
+            "CREATE TABLE IF NOT EXISTS " +
+                    INDEXED_FIXTURE_PATHS_TABLE +
+                    " (" + INDEXED_FIXTURE_PATHS_COL_NAME + " UNIQUE" +
+                    ", " + INDEXED_FIXTURE_PATHS_COL_BASE +
+                    ", " + INDEXED_FIXTURE_PATHS_COL_CHILD + ");";
+
+    public static IndexedFixtureIndex lookupIndexedFixturePaths(SQLiteDatabase db,
+                                                                String fixtureName) {
+        Cursor c = db.query(INDEXED_FIXTURE_PATHS_TABLE,
+                new String[]{INDEXED_FIXTURE_PATHS_COL_BASE, INDEXED_FIXTURE_PATHS_COL_CHILD, INDEXED_FIXTURE_PATHS_COL_ATTRIBUTES},
+                INDEXED_FIXTURE_PATHS_COL_NAME + "=?", new String[]{fixtureName}, null, null, null);
         try {
             if (c.getCount() == 0) {
                 return null;
             } else {
                 c.moveToFirst();
-                return Pair.create(
-                        c.getString(c.getColumnIndexOrThrow(IndexedFixturePathsConstants.INDEXED_FIXTURE_PATHS_COL_BASE)),
-                        c.getString(c.getColumnIndexOrThrow(IndexedFixturePathsConstants.INDEXED_FIXTURE_PATHS_COL_CHILD)));
+                TreeElement attrsHolder = null;
+                byte[] attrsBlob = c.getBlob(c.getColumnIndexOrThrow(INDEXED_FIXTURE_PATHS_COL_ATTRIBUTES));
+                if (attrsBlob != null) {
+                    attrsHolder = SerializationUtil.deserialize(attrsBlob, TreeElement.class);
```

Comment:
I think we might need to have the prototype factory available here. I don't think it's relevant today, but it's best to avoid ever deserializing without it available.

Reply:
the `deserialize` currently makes use of the prototype factory internally.

Reply:
Ah, got it. I thought this path used the "default" one, but it looks like it pulls it live from the session